### ISSUE TITLE
[Issue]: home screen is repeating gets many times per second

### ISSUE THREAD

**Issue by sevenrats** [jellyflix-app/jellyflix#57]
### Please describe your bug

I thought I was having performance issues with the VM but now that I've got some more logging on the server side its clear that something is wrong with the home screen. note that these are all the exact same 3 requests being repeated over and over.  presumeably these extra 15 requests per second or so are getting in the way.

### Jellyflix Version

0.9.13

### Jellyfin Version

10.8.13

### Environment

```markdown
- OS:debian
- OS Version:trixie
- Playback Method: not there yet
```

**Comment by sevenrats** [comment]
![image](https://github.com/jellyflix-app/jellyflix/assets/79296037/c94b76d4-54e6-4ec0-87ce-f8b9c3a6680e)

**Comment by sevenrats** [comment]
looking in the flutter inspector suggests that the app is request the user library views up to twenty times per second, and never stop doing so until the view is left.

**Comment by sevenrats** [comment]
the issue appears to originate from home_screen.dart:146:43

**Comment by sevenrats** [comment]
this issue is fundamentally caused by the lack of comprehensive state checks in paginated item carousel line 54.
it is not the only infinite get recursion in the code though. another occurs during playback and is no doubt the cause of my performance issues.

**Comment by jdk-21** [comment]
I added another check, which should fix this issue in 64ef283bb06ddf3dfb67c506d0ca0c760b86ea12

**Comment by sevenrats** [comment]
awesome. i will pull it down.  as i mentioned in chat, the other loop is here: 
package:jellyflix/components/genre_banner.dart:76:68
